address/address.go

```diff
@@ -118,14 +119,18 @@ type Tap struct {
 	// assetGen is the receiving asset's genesis metadata which directly
 	// maps to its unique ID within the Taproot Asset protocol.
 	assetGen asset.Genesis
+
+	// ProofCourierAddr is the address of the proof courier that will be
+	// used to distribute related proofs for this address.
+	ProofCourierAddr *url.URL
 }
 
 // New creates an address for receiving a Taproot asset.
 func New(genesis asset.Genesis, groupKey *btcec.PublicKey,
 	groupSig *schnorr.Signature, scriptKey btcec.PublicKey,
 	internalKey btcec.PublicKey, amt uint64,
 	tapscriptSibling *commitment.TapscriptPreimage,
-	net *ChainParams) (*Tap, error) {
+	net *ChainParams, proofCourierAddr *url.URL) (*Tap, error) {
```

Comment:
Non-blocking, but starting to be a lot of values, can maybe split out into a new sub-struct for the params? I think the genesis+group key stuff can use `GroupKeyInfo`, but maybe to wait until we get the v1 asset stuff in.

Reply:
I'll add a TODO in `New` function doc.